### PR TITLE
Add gzip to nginx conf

### PR DESCRIPTION
**Pull request by MonPote**
**Component**: ui

**Context**: 
Currently the browser need to download 3Mb of JS. We can drastically decrease this number by compression thanks to nginx.

**Summary**:

- Enable gzip

This will decrease the size of the file the client need to download
In the current version we have ~7x improvement.
From 3.17Mb to 0.470Mb send to the client.

**Acceptance criteria**: 

You can reach the UI and play with it. Everything should be the same.
When you load the UI for the first time, open the console before and on the network tab you should see for the JS files something like [this](https://puu.sh/EcU7g/4a22ffde50.png) instead of something like [this](https://puu.sh/EcU6m/9ff1243eed.png)

### CODE REVIEW

File: ui/conf/nginx.conf [hunk 1]
```diff
@@ -12,13 +28,14 @@ server {
         try_files $uri $uri/ /index.html;
     }
 
+    # Provide the ConfigMaps generated by MetalK8s
```
Comment (slaperche-scality) [inline ui/conf/nginx.conf:30]: not related to gzip

File: ui/conf/nginx.conf [hunk 2]
```diff
@@ -12,13 +28,14 @@ server {
...
     location /config.json {
         alias /etc/metalk8s/ui/config.json;
     }
 
+    # Use the theme provided by the customer
```
Comment (slaperche-scality) [inline ui/conf/nginx.conf:35]: ditto (should be in a dedicated commit "add comments" or something like that.

File: ui/conf/nginx.conf [hunk 3]
```diff
@@ -2,8 +2,24 @@ server {
     listen       80;
     server_name  localhost;
 
-    #charset koi8-r;
-    #access_log  /var/log/nginx/host.access.log  main;
+    # https://www.digitalocean.com/community/tutorials/how-to-increase-pagespeed-score-by-changing-your-nginx-configuration-on-ubuntu-16-04
```
Comment (slaperche-scality) [inline ui/conf/nginx.conf]: I would rather link the official nginx documentation here: https://nginx.org/en/docs/http/ngx_http_gzip_module.html instead of a "random" blog.

File: ui/conf/nginx.conf [hunk 4]
```diff
@@ -2,8 +2,24 @@ server {
...
+    gzip on;
+    gzip_comp_level    5;
+    gzip_min_length    256;
+    gzip_proxied       any;
```
Comment (slaperche-scality) [inline ui/conf/nginx.conf]: Do we need this one?

File: ui/conf/nginx.conf [hunk 5]
```diff
@@ -2,8 +2,24 @@ server {
...
+    gzip_vary          on;
```
Comment (slaperche-scality) [inline ui/conf/nginx.conf]: Ditto, do we need this.

Seems useful to handle a very rare case so not sure it's worth it.

**Review by slaperche-scality — DISMISSED** [review]

File: ui/config-overrides.js [hunk 6]
```diff
@@ -1,3 +1,13 @@
-const { useBabelRc, override, useEslintRc } = require('customize-cra');
+const {
```
Comment (slaperche-scality) [inline ui/config-overrides.js:1]: Hum, wrong commit?

Looks like this change belongs to "Add compression-webpack-plugin to create gziped file durint the build" instead of "Add gzip to nginx conf".

File: ui/conf/nginx.conf [hunk 7]
```diff
@@ -2,8 +2,23 @@ server {
...
-    #charset koi8-r;
-    #access_log  /var/log/nginx/host.access.log  main;
+    # https://nginx.org/en/docs/http/ngx_http_gzip_module.html
+    gzip on;
+    gzip_static on;
+    gzip_comp_level    5;
+    gzip_min_length    256;
+
+    gzip_types
+    application/javascript
```
Comment (NicolasT) [inline ui/conf/nginx.conf]: Consider indenting these a bit more, for clarity.

File: ui/config-overrides.js [hunk 8]
```diff
@@ -1,3 +1,13 @@
...
+  useEslintRc,
+  addWebpackPlugin
+} = require('customize-cra');
+const CompressionPlugin = require('compression-webpack-plugin');
 
-module.exports = override(useBabelRc(), useEslintRc());
+module.exports = override(
+  useBabelRc(),
+  useEslintRc(),
+  addWebpackPlugin(new CompressionPlugin())
```
Comment (NicolasT) [inline ui/config-overrides.js]: Add a `,` at the end so later commits won't change this line for no good reason.

**Review by NicolasT — CHANGES_REQUESTED** [review]

File: ui/conf/nginx.conf (hunk 7 above)
Comment (MonPote) [reply ui/conf/nginx.conf]: Yep, I just add an indent.

File: ui/config-overrides.js (hunk 8 above)
Comment (MonPote) [reply ui/config-overrides.js]: I just add the rule on .prettierc that will also enforce that rule in the entire project :)

**Review by NicolasT — APPROVED** [review]
OK. However:

- Could you file a ticket for code/bundle-splitting.
- Could you file a ticket for configuring Webpack's bundle-size-limits-becomes-a-build-error feature, so CI catches bundle growth accordingly?

Both of these could be handled in the same PR of course.

**Comment by MonPote** [comment]
@bert-e approve

**Comment by NicolasT** [comment]
For reference, https://webpack.js.org/configuration/performance/#performancehints

**Comment by MonPote** [comment]
> For reference, https://webpack.js.org/configuration/performance/#performancehints



> OK. However:
> 
> * Could you file a ticket for code/bundle-splitting.
> * Could you file a ticket for configuring Webpack's bundle-size-limits-becomes-a-build-error feature, so CI catches bundle growth accordingly?
> 
> Both of these could be handled in the same PR of course.

Ok let me do that.
